crates/cheatnet/src/state.rs

```diff
@@ -27,11 +32,42 @@ pub struct ExtendedStateReader {
     pub fork_state_reader: Option<ForkStateReader>,
 }
 
+pub trait BlockInfoReader {
+    fn get_block_info(&self) -> StateResult<CheatnetBlockInfo>;
+}
+
+impl BlockInfoReader for ExtendedStateReader {
+    fn get_block_info(&self) -> StateResult<CheatnetBlockInfo> {
+        if self.fork_state_reader.is_some() {
+            return self.fork_state_reader.as_ref().unwrap().get_block_info();
```

Comment:
`if let Some(fork_state_reader) = ...` would let us avoid the `unwrap`

Reply:
`if let Some(fork_state_reader) = self.fork_state_reader` will require for `self.fork_state_reader` to be moved, which is forbidden.

Reply:
What about `if let Some(ref fork_state_reader) = ...`?

Reply:
good point! done